### PR TITLE
[@unimodules][screen-orientation] remove fbjs dependency, add replacement methods to Platform

### PR DESCRIPTION
**Pull request by cruzach**
# Why

see https://github.com/expo/expo/pull/11396 for full explanation on why to remove `fbjs`

# How

Added replacements for `canUseViewport` and `canUseEventListeners`, and used those in `expo-screen-orientation`

# Test Plan

added tests

### CODE REVIEW

File: packages/expo-screen-orientation/src/ExpoScreenOrientation.web.ts [hunk 1]
```diff
@@ -33,9 +32,9 @@ const OrientationWebToAPI: {
 
 declare const window: Window;
 
-const { screen } = canUseViewport && window;
+const { screen }: any = Platform.canUseViewport && window;
```
Comment (lukmccall) [inline packages/expo-screen-orientation/src/ExpoScreenOrientation.web.ts:35]: I don't like `any` type here. Can we change it to something else?

Comment (cruzach) [reply packages/expo-screen-orientation/src/ExpoScreenOrientation.web.ts]: yeah, updated here https://github.com/expo/expo/pull/11398/commits/51a0b5d5494de3712fe8985568d37b5e4ae3093e

**Review by lukmccall — APPROVED** [review]
LGTM 💪💪💪

**Review by tsapeta — APPROVED** [review]

**Review by sjchmiela — APPROVED** [review]
Sweet! 💯

**Pull request merged**